### ISSUE TITLE
[Question] How does the client detection work?

### ISSUE THREAD

**Issue by Tone866** [RobinLinus/snapdrop#329]
Hi,
I was wondering how a snapdrop instance notice other devices on the local network.
A network-scan is probably too slow.
There are not many comments in the code, so I thought I just ask.
Thanks!

**Comment by fm-sys** [comment]
It uses a stun server, see... 

https://github.com/RobinLinus/snapdrop/blob/5389464605fe7f673e05034aa89dfc708d05bc4d/client/scripts/network.js#L514-L519

**Comment by Tone866** [comment]
hm, stun is typical used to connect clients behind different NATs. It also could be used for clients behind the same NAT ok, but how snapdrop detects that they are behind the same NAT?
If the public IP is the same?

**Comment by Bellisario** [comment]
Yes, the server puts clients with the same ip in a room and makes everyone in the same room can watch others.
Look here:
https://github.com/RobinLinus/snapdrop/blob/5389464605fe7f673e05034aa89dfc708d05bc4d/server/index.js#L69-L78

**Comment by Tone866** [comment]
Ok thanks!
So that would meen if my isp uses NAT, all other customers behind the same IP would be in the same room with me?

**Comment by Bellisario** [comment]
No, because your router have an ip that is used to makes requests on the web: all sites that want to know your ip address, can use your router public ip address. ISP is an intermediary that, for example, connects a router to a server, and "says" that the request is made by the router and that if he wants to respond, he needs to say he wants to send data to this ip. ISP exists because is impossible my by a router to reach a server located in the other side of the world (but also another router in the same room), and ISP do this for us: we can define  them as a "postman".

**Comment by Tone866** [comment]
I meant if the ISP is using Carrier-grade NAT, so one public IP is used for multiple customers. 
So even my routers WAN-IP is private.
